src/frontend/src/lib/services/fiat.services.ts

```diff
@@ -0,0 +1,39 @@
+import { ETHEREUM_TOKEN_ID } from '$lib/constants/tokens.constants';
+import { Erc20PriceByContract, EthPrice } from '$lib/providers/coingecko.providers';
+import { balancesStore } from '$lib/stores/balances.store';
+import { fiatStore } from '$lib/stores/fiat.store';
+import { toastsError } from '$lib/stores/toasts.store';
+import { get } from 'svelte/store';
+
+export const loadFiatBalance = async (): Promise<{ success: boolean }> => {
+	const balances = get(balancesStore);
+
+	// https://github.com/microsoft/TypeScript/issues/1863
+	// @ts-ignore
+	const { [ETHEREUM_TOKEN_ID]: ethBalance, ...erc20Tokens } = balances;
+
+	try {
+		const ethPrice = await EthPrice();
+		fiatStore.set({ tokenId: ETHEREUM_TOKEN_ID, fiatBalance: ethPrice! });
+	} catch (err: unknown) {
+		fiatStore.reset();
+		toastsError({
+			msg: { text: 'Error while fetching Ethereum price.' },
+			err
+		});
+	}
+
+	try {
+		const erc20Prices = await Erc20PriceByContract(Object.getOwnPropertySymbols(erc20Tokens));
```

Comment:
why `Object.getOwnPropertySymbols`? this is uncommon in regard to the existing code base